### PR TITLE
Example/get all links #721

### LINKED ISSUE #721: How to get all the links from a page
Based on https://github.com/kblok/puppeteer-sharp/issues/708

### PR DESCRIPTION
**Pull request by bdurrani**
closes #721 

I used [this guide here](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) for some conventions to try to keep things consistent as I make more examples.

I only tested this example on Windows for now.

### CODE REVIEW

File: samples/get-all-links/Program.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            var options = new LaunchOptions { Headless = true };
+            Console.WriteLine("Downloading chromium");
+            await new BrowserFetcher().DownloadAsync(BrowserFetcher.DefaultRevision);
+            Console.WriteLine("Navigating to google.com");
+
+            using (var browser = await Puppeteer.LaunchAsync(options))
+            {
+                using (var page = await browser.NewPageAsync())
+                {
```
Comment (kblok) [inline samples/get-all-links/Program.cs]: We tend to code these 2 usings like this:
```cs 
using (var browser = await Puppeteer.LaunchAsync(options))
using (var page = await browser.NewPageAsync())
{
}
```

**Comment by kblok** [comment]
Let's add [here](https://github.com/kblok/puppeteer-sharp/blob/master/docfx_project/examples/toc.yml#L4) and item call "How to extract all the links from a page" with a file "Page.EvaluateExpressionAsync.GetAllLinks.md".
Make a copy of https://github.com/kblok/puppeteer-sharp/blob/master/docfx_project/examples/Page.ScreenshotAsync.md:

 * Add yourself as the contributor.
 * Explain the problem.
 * Copy there the code from [line 16](https://github.com/kblok/puppeteer-sharp/pull/729/files#diff-39be4281f91fee4b1983e88ba98259bbR16)

**Comment by bdurrani** [comment]
@kblok updated as per your feedback

File: docfx_project/examples/Page.EvaluateExpressionAsync.GetAllLinks.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+You need to get all links from a page.
+
+## Solution
+
+Use `Page.EvaluateExpressionAsync` to evaluate javascript in the context of the browser
+and return the `href` associated with the hyperlink tag.
+
+```cs
+    using (var browser = await Puppeteer.LaunchAsync(options))
```
Comment (kblok) [inline docfx_project/examples/Page.EvaluateExpressionAsync.GetAllLinks.md]: We can remove the extra indentation

File: docfx_project/examples/toc.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 - name: Basic Examples
   items:
-    - name: How to take screenshots and save it as a file
-      href: Page.ScreenshotAsync.md
-      
\ No newline at end of file
+      - name: How to take screenshots and save it as a file
+        href: Page.ScreenshotAsync.md
+      - name: How to extract all the link from a page
+        href: Page.EvaluateExpressionAsync.GetAllLinks.md
```
Comment (kblok) [inline docfx_project/examples/toc.yml]: Careful here, we need to use spaces and only 2 spaces

**Comment by bdurrani** [comment]
done!

**Comment by kblok** [comment]
Awesome! Is this still WIP? Or is it ready to be merged?

**Comment by bdurrani** [comment]
Its ready. I tested this on Windows 10.

**Pull request merged**